### PR TITLE
Add edit function

### PR DESCRIPTION
**Pull request by richherbs**

### CODE REVIEW

File: includes/displayBikes.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ * @return void
+ */
+function printCards (array $allBikes){
+    $allCards = '';
+
+    foreach($allBikes as $bike){
+        $deleted = $bike['deleted'];
+        if($deleted){
+            continue;
+        }
```
Comment (ashleycoles) [inline includes/displayBikes.php]: I'd prefer this check to happen in MySQL instead

File: tests/includes/displayBikes.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            $case = printCards($input);
+            $this->assertEquals($expected, $case);
+        }
+
+        public function testFailurePrintCards(){
+            $expected = "<div class='card'><img class='bike-image' src='2' alt='1 1 bike'><div class=\"card-info-container\"><section>Make: 1</section><section>Model: 1</section><section>Discipline: 1</section><section>Wheel Size: 1</section></div><div class='card-button-container'><button>Edit Bike</button><button>Delete Bike</button></div></div>";
+            $input = [['pic_url'=>1,'brand_name'=>1,'model'=>1,'discipline_name'=>1,'wheel_diameter'=>1]];
+            $case = printCards($input);
+            $this->assertNotEquals($expected, $case);
+        }
```
Comment (ashleycoles) [inline tests/includes/displayBikes.php]: Doesn't look like these tests have been updated to include the delete flag and ID you're now using for story 3.

File: tests/includes/populateSelectors.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            $case = populateSelector($input);
+            $this->assertNotEquals($expected, $case);
+        }
+
+        public function testMalformedPopulateSelector(){
+            $expected = '';
+            $input = 'string';
+            $this->expectException(TypeError::class);
+            $case = populateSelector($input);
+        }
```
Comment (ashleycoles) [inline tests/includes/populateSelectors.php]: populateSelector now takes 2 params

File: index.php [hunk 4]
```diff
@@ -1,70 +1,20 @@
...
-                <section>Wheel Size: 29er</section>
-            </div>
-            <div class='card-button-container'>
-                <button>Edit Bike</button>
-                <button>Delete Bike</button>
-            </div>
-        </div>
+        <?php  
+            $bikes = getBikesFromDB($db);
+            echo printCards($bikes);
```
Comment (ashleycoles) [inline index.php:13]: What if there are no bikes found?

**Review by ashleycoles — CHANGES_REQUESTED** [review]
Looking really good Rich! A couple of comments in the review for you, I'd also like you to spend a bit of time improving the view on mobile (just tweaking the columns). I know responsive isn't in your team definition of done, but I will be suggesting it during retro!

**Review by ashleycoles — APPROVED** [review]
Spot on, great job!

File: index.php (hunk 4 above)
Comment (richherbs) [reply index.php:13]: it does the function but nothing happens

**Pull request merged**